src/inmanta/plugins.py

```diff
@@ -543,6 +553,13 @@ def wrapper(self, *args: object, **kwargs: object) -> Any:
             if name is None:
                 name = fnc.__name__
 
+            ns_parts = str(fnc.__module__).split(".")
+            ns_parts.append(name)
+            if ns_parts[0] != const.PLUGINS_PACKAGE:
+                raise Exception("All plugin modules should be loaded in the %s package" % const.PLUGINS_PACKAGE)
+
+            fnc.plugin_name = "::".join(ns_parts[1:])
```

Comment:
We discussed it with @sanderr and we both preferred the current option. My only argument is that I find it easier to read but I don't really mind changing it :)

Reply:
That was the previous implementation. We then came up with this new idea and considered it slightly cleaner (though the current implementation still needs a bit of refinement): by attaching metadata to the function returned by the `plugin` decorator, we keep the lookup simple: we only need to maintain one mapping, and we only need to convert between Python and inmanta names once.
Essentially, we just pass metadata over the decorators directly rather than having to extend the registry and lookup class to handle Python names.
We decided to give it a go and see what you thought. Do you prefer the old mechanism?

Reply:
@wouterdb after the latest changes I quite like the current implementation. It definitely feels simpler to me than the previous one.

Reply:
I like it as well, if it is correct. It changes an internal API, so it might break the plugins, but I assume you checked this?

Reply:
I did. As far as I can see there should be no fallout. We'll only know for sure the morning after it gets merged of course.

Reply:
I tested it with a little module and model I've been using to develop this feature. Should I have tested it more than that if so, how?

Reply:
I don't think so. It's just a question of what, if anything, we have overlooked. For example, pytest-inmanta does some magic with plugin loading. I've checked the code and I think it shouldn't be affected by this change. But the only way to really know for sure with this many components playing together is to run the full set of integration tests.